### PR TITLE
[CHK-3055] Add span for send payment result monitoring

### PR DESCRIPTION
**Pull request by ciuffagianluca**
Add span to track send payment result in kibana

#### List of Changes

Moved inside the handler the succes case to track for the send payment result processing in order to have more info about the transaction on which the send payment result is about.
Introducing some info in AlreadyProcessedException to track the info about error flow in send payment result
Introducing new Processing exception to track edge case of send payment result sent when transaction is in ACTIVATED state

#### Motivation and Context

Improve ELK monitoring

#### How Has This Been Tested?

Tests have been run in dev env

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/java/it/pagopa/transactions/commands/handlers/v2/TransactionRequestUserReceiptHandler.java [hunk 1]
```diff
@@ -73,7 +83,30 @@ public Mono<BaseTransactionEvent<?>> handle(TransactionAddUserReceiptCommand com
...
+                                    .getTransactionGatewayAuthorizationRequestedData()).getWalletInfo() != null;
+                    return Mono.error(
+                            new AlreadyProcessedException(
+                                    t.getTransactionId(),
+                                    t.getTransactionAuthorizationRequestData().getPspId(),
+                                    t.getTransactionAuthorizationRequestData().getPaymentTypeCode(),
+                                    t.getClientId().name(),
+                                    isWalletPayment,
+                                    // TODO Qui ci va il codice di risposta della send payment result ? L'error code
+                                    // sarà sempre empty() ?
```
Comment (giovanniberti) [inline src/main/java/it/pagopa/transactions/commands/handlers/v2/TransactionRequestUserReceiptHandler.java:102]: `Option.empty()` is ok

File: src/main/java/it/pagopa/transactions/commands/handlers/v2/TransactionRequestUserReceiptHandler.java [hunk 2]
```diff
@@ -104,62 +137,111 @@ public Mono<BaseTransactionEvent<?>> handle(TransactionAddUserReceiptCommand com
...
+                                            language,
+                                            addUserReceiptRequestDto.getPaymentDate().toZonedDateTime().toString()
                                     )
-                                            .doOnNext(
-                                                    queueResponse -> log.info(
+                            ),
+                            tx
+                    );
+                }).flatMap(
+                        event_tx -> userReceiptAddedEventRepository.save(event_tx.getT1())
```
Comment (giovanniberti) [inline src/main/java/it/pagopa/transactions/commands/handlers/v2/TransactionRequestUserReceiptHandler.java:193]: If you want to avoid doing `event_tx.getT1()` and so on you could refactor this into

```suggestion
                        TupleUtils.function((event, tx) -> userReceiptAddedEventRepository.save(event)
```

File: src/main/java/it/pagopa/transactions/configurations/UpdateTransactionStatusTracerUtilsConfig.java [hunk 3]
```diff
@@ -1,5 +1,6 @@
 package it.pagopa.transactions.configurations;
 
+import io.opentelemetry.api.trace.Tracer;
```
Comment (giovanniberti) [inline src/main/java/it/pagopa/transactions/configurations/UpdateTransactionStatusTracerUtilsConfig.java:3]: Stray import?

File: src/test/java/it/pagopa/transactions/controllers/v1/TransactionsControllerTest.java [hunk 4]
```diff
@@ -1184,20 +1185,117 @@ void shouldReturnUnprocessableEntityInSendPaymentResultForUncaughtError() {
...
+        String contextPath = "auth-requests";
         /*
-         * UpdateTransactionStatusTracerUtils.NodoStatusUpdate expectedStatusUpdateInfo
-         * = new UpdateTransactionStatusTracerUtils.NodoStatusUpdate(
+         * UpdateTransactionStatusTracerUtils.PaymentGatewayStatusUpdate
+         * expectedStatusUpdateInfo = new
+         * UpdateTransactionStatusTracerUtils.PaymentGatewayStatusUpdate(
          * UpdateTransactionStatusTracerUtils.UpdateTransactionStatusOutcome.
-         * INVALID_REQUEST );
+         * INVALID_REQUEST, expectedTrigger, Optional.empty(), Optional.empty() );
```
Comment (giovanniberti) [inline src/test/java/it/pagopa/transactions/controllers/v1/TransactionsControllerTest.java:1285]: Stray comment?

File: src/test/java/it/pagopa/transactions/utils/UpdateTransactionStatusTracerUtilsTest.java [hunk 5]
```diff
@@ -0,0 +1 @@
+
```
Comment (giovanniberti) [inline src/test/java/it/pagopa/transactions/utils/UpdateTransactionStatusTracerUtilsTest.java:1]: empty file? :)

**Review by giovanniberti — APPROVED** [review]

File: src/main/java/it/pagopa/transactions/controllers/v1/TransactionsController.java [hunk 6]
```diff
@@ -264,6 +287,56 @@ public Mono<ResponseEntity<AddUserReceiptResponseDto>> addUserReceipt(
                 );
     }
 
+    /**
+     * This method maps input throwable to proper {@link UpdateTransactionStatusTracerUtils.UpdateTransactionStatusOutcome} enumeration
```
Comment (pietro-tota) [inline src/main/java/it/pagopa/transactions/controllers/v1/TransactionsController.java:291]:
```suggestion
     * This method maps input throwable to proper {@link UpdateTransactionStatusTracerUtils.SendPaymentResultOutcomeInfo} enumeration
```

File: src/main/java/it/pagopa/transactions/exceptions/ProcessingErrorException.java [hunk 7]
```diff
@@ -0,0 +1,9 @@
+package it.pagopa.transactions.exceptions;
+
+public class ProcessingErrorException extends RuntimeException {
```
Comment (pietro-tota) [inline src/main/java/it/pagopa/transactions/exceptions/ProcessingErrorException.java:3]: This exception is mapped to 422 HTTP response code by code [here](https://github.com/pagopa/pagopa-ecommerce-transactions-service/blob/0bce94e59031680a7dec2d6ed9ed80928483bebf/src/main/java/it/pagopa/transactions/controllers/v1/TransactionsController.java#L501-L522). Is this wanted or is a side-effect?

File: src/main/java/it/pagopa/transactions/commands/handlers/v2/TransactionRequestUserReceiptHandler.java [hunk 8]
```diff
@@ -74,38 +74,56 @@ public Mono<BaseTransactionEvent<?>> handle(TransactionAddUserReceiptCommand com
                 );
 
         Mono<it.pagopa.ecommerce.commons.domain.v2.TransactionClosed> alreadyProcessedError = transaction
```
Comment (giovanniberti) [inline src/main/java/it/pagopa/transactions/commands/handlers/v2/TransactionRequestUserReceiptHandler.java:76]: Just food for thought, maybe better for another PR: given that this chain is quite long and filled with conditionals, what do you think about making a method that computes the result?

**Review by giovanniberti — APPROVED** [review]
:rocket:

File: src/main/java/it/pagopa/transactions/commands/handlers/v2/TransactionRequestUserReceiptHandler.java (hunk 8 above)
Comment (ciuffagianluca) [reply src/main/java/it/pagopa/transactions/commands/handlers/v2/TransactionRequestUserReceiptHandler.java:76]: I'll open an issue to track this

File: src/main/java/it/pagopa/transactions/exceptions/ProcessingErrorException.java (hunk 7 above)
Comment (ciuffagianluca) [reply src/main/java/it/pagopa/transactions/exceptions/ProcessingErrorException.java:3]: It is wanted. 
In this exception I basically want to track this edge case (send payment result for a transction without an authorization, really not possibile). Is this a little bit different from the Already processed exception: here we map the fact that we don't expect this call from nodo simply because the transaction isn't yet authorized. So the 422 response code is ok.
